src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java

```diff
@@ -115,7 +117,8 @@ void persist(Map<String, String> eventHeaders, NotifyContextRequest notification
                 
                 // create the collection at this stage, if the data model is collection-per-attribute
                 if (dataModel == DataModel.COLLECTIONPERATTRIBUTE) {
-                    collectionName = buildCollectionName(fiwareServicePaths[i], destinations[i], attrName);
+                    collectionName = buildCollectionName(dbName, fiwareServicePaths[i], destinations[i], attrName,
+                            false, entityId, entityType, fiwareService);
                     backend.createCollection(dbName, collectionName);
```

Comment:
`createCollection()` throws a exception if something goes wrong. This exception, as others, are propagated to `OrionSink` base class, which handles them in the appropriate way.